src/NosCore.GameObject/Group.cs

```diff
@@ -59,31 +59,39 @@ public Group(GroupType type)
 
         public PinitPacket GeneratePinit()
         {
-            var i = 0;
+            var i = 1;
+
+            if (IsEmpty) return new PinitPacket { GroupSize = 0 };
 
             return new PinitPacket
             {
                 GroupSize = Count == 1 ? 0 : Count,
-                PinitSubPackets = Values.Select(s => s.Item2.GenerateSubPinit(Count == 1 ? i : ++i)).ToList()
+                PinitSubPackets = Values.OrderBy(s => s.Item1).Select(s => s.Item2.GenerateSubPinit(Count == 1 ? i : ++i)).ToList()
             };
         }
 
-        public List<PstPacket> GeneratePst()
+        public List<PstPacket> GeneratePst(long visualId)
         {
-            var i = 0;
+            var i = 1;
+
+            INamedEntity avoid(INamedEntity namedEntity)
```

Comment:
no

Reply:
Any other method to do this? xD, sorry for being a noob

Reply:
just keep the code as it was and just where != namedEntity

Reply:
With as it was you mean as I done it or how it had been before I touched it?. I mean, the code that you want to keep used to update the "grouporder" while making the PstPacket. This means that if I put a ".where" and remove the "avoid" that will cause the order to mess, messing up the info NosTale gets.

Reply:
You avoid is just doing a ++i which was already done in the code

Reply:
But if we use ".where" it will only do "++i" when not sending the info. Let me explain. Grouporder needs to prevail even if you don't send that "pst", so, doing "++i" only when sending that exact "pst" will mess up the order

Reply:
So you just want the order not i

Reply:
This are the order for pinit and pst (If I don't remember badly) -> 0 = companion | 1 = pet | 2 = Party Master | 3 = Party member (next party master if Party Master leaves the party) | 4 = Party member. I use "i" as the grouporder (because it was used too in the original code) and the "lastId" as the Order